source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -141,7 +142,21 @@ void AuthenticatorImpl::startVerify() {
 
   jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
   ENVOY_LOG(debug, "{}: Parse Jwt {}", name(), curr_token_->token());
-  Status status = jwt_->parseFromString(curr_token_->token());
+  Status status;
+
+  if (!token_cache_->lookupTokenCacheData(curr_token_->token(), *jwt_, status)) {
```

Comment:
Some problems with this cache code
1) jwt ownership is not clear. 
2) I only see a jwt parsing step is skipped with cache hit.  I don't see how "verify" step is skipped with a cache hit case.

Reply:
PTAL